rtf_vis_tool/src/Components/Bundle_Adj_PC.js

```diff
@@ -0,0 +1,116 @@
+import React, {useEffect, useState, useRef} from "react";
+import {Canvas, extend} from "react-three-fiber";
+import {OrbitControls} from 'three/examples/jsm/controls/OrbitControls';
+import '../stylesheets/Bundle_Adj_PC.css';
+
+//Loading Helper Components
+import CoordinateGrid from './CoordinateGrid';
+import OrbitControlsComponent from './OrbitControlsComponent';
+import PointMesh from './PointMesh';
+
+//allows user to orbit and pan around the point cloud
+extend({OrbitControls});
+
+//Point Cloud Renderer Component
+//Spawned once the 'SfMData' node is clicked
+const Bundle_Adj_PC = (props) => {
+    const [pointCloudRaw, setPointCloudRaw] = useState([]);
+    const [pointCloudJSX, setPointCloudJSX] = useState([]);
+    const [showCoordGrid, setShowCoordGrid] = useState(true);
+    const [pointRadius, setPointRadius] = useState(0.15);
+    const pointSizeArr = [pointRadius];         //point size simply defined by the radius, initialized as 0.15
+
+    //render points3D.txt from COLMAP ba_output directory
+    useEffect(() => {
+        //fetch the specific file from the public directory
+        fetch('results/ba_output/points3D.txt')
```

Comment:
where is this relative to?

Reply:
this is relative to `rtf_vis_tool/public` directory. For reading local text files, I have to use the `fetch()` function, which can only read files that are within `rtf_vis_tool/public`. But for JSON files, I can nicely import them (which is what I do with `data_association_metrics.json`)